examples/cholesky.plv

```diff
@@ -0,0 +1,103 @@
+import prelude;
+
+-- "Matrix Computations" 4th ed., Golub and Van Loan.  Algorithm
+-- 4.1.2, the LDL^T decomposition for a symmetric, positive-definite
+-- matrix.
+--
+-- On success, the argument A now contains the lower-triangular factor
+-- L and the diagonal vector D.
+ldlt {m}
+  (inout A :: double[m, m])
+  :: s8 := (
+
+  v :: double[m];
+
+  for j in 0:m -> (
+    -- Compute v[1:j].
+    for i in 0:j -> (
+       v[i] <- A[j, i] * A[i, i];
+    );
+    v[j] <- A[j, j] - A[j, 0:j] * v[0:j];
+
+    -- Store d[j].
+    A[j, j] <- v[j];
+
+    -- Compute L[j+1:n, j].
+    A[j+1:m, j] <- (A[j+1:m, j] - A[j+1:m, 0:j]*v[0:j]) / v[j];
+  );
+
+  return 0;
+);
+
+-- "Advanced Kalman Filtering, Least-Squares and Modeling.", Bruce
+-- P. Gibbs.  Algorithm 10.2-2, the UDU^T decomposition for a
+-- symmetric, positive-definite matrix.
+--
+-- On exit, the argument A now contains the upper-triangular factor U
+-- and the diagonal vector D.
+udut {m}
+  (inout A :: double[m, m])
+  :: Void := (
+  for j in m-1:0:-1 -> (  -- column loop
```

Comment:
for someone comparing to the book, do you think `for j in m-1..1:-1 -> ...` is clearer?